src/main/groovy/com/morpheus/scvmm/sync/VirtualMachineSync.groovy

```diff
@@ -0,0 +1,490 @@
+                            if (osType && currentServer.serverOs != osType) {
+                                currentServer.serverOs = osType
+                                currentServer.osType = currentServer.serverOs?.platform?.toLowerCase()
+                                currentServer.platform = osType?.platform
+                                planInfoChanged = true
+                                save = true
+                            }
+                            //plan
+                            ServicePlan plan = SyncUtils.findServicePlanBySizing(availablePlans, currentServer.maxMemory, currentServer.maxCores,
+                                    null, fallbackPlan, currentServer.plan, currentServer.account, [])
+                            if (currentServer.plan?.id != plan?.id) {
+                                currentServer.plan = plan
+                                planInfoChanged = true
+                                save = true
+                            }
+                            def volumeChanged = false
+                            if (masterItem.Disks) {
+                                if (currentServer.status != 'resizing' && currentServer.status != 'provisioning') {
+                                    volumeChanged = syncVolumes(currentServer, masterItem.Disks)
+                                    if (volumeChanged == true) {
+                                        planInfoChanged = true
+                                        save = true
+                                    }
+                                }
+                            }
+                            log.debug ("updateMatchedVirtualMachines: save: ${save}")
+                            if (save) {
+                                saves << currentServer
+                            }
+                        } catch (ex) {
+                            log.error("Error Updating Virtual Machine ${currentServer?.name} - ${currentServer.externalId} - ${ex}", ex)
+                        }
+                    }
+                } catch (Exception e) {
+                    log.error "Error in updating stats: ${e.message}", e
+                }
+            }
+            if (saves) {
+                context.async.computeServer.bulkSave(saves).blockingGet()
+            }
+        } catch (Exception e) {
+            log.error "Error in updating virtual machines: ${e.message}", e
+        }
+    }
+
+    def removeMissingVirtualMachines(List<ComputeServerIdentityProjection> removeList) {
+        log.debug("removeMissingVirtualMachines: ${cloud} ${removeList.size()}")
+        def removeItems = context.services.computeServer.list(
+                new DataQuery().withFilter("id", "in", removeList.collect { it.id })
+                        .withFilter("computeServerType.code", 'scvmmUnmanaged')
+        )
+        context.async.computeServer.remove(removeItems).blockingGet()
```

Comment:
Why is a list of the full objects being loaded instead of just using the `ComptueServerIdentityProjection`?

Reply:
changed list to listIdentityProjections